Update task list documentation and example (#38245)
woocommerce/woocommerce: plugins/woocommerce-admin/docs/features/onboarding-tasks.md

  }

  public function get_time() {
    return __( '2 minutes', 'woocommerce' );
  }
}
```

You can then add the task to the task list by calling the `add_task` method on the `TaskLists` class.

```php
use Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists;

TaskLists::add_task(
  'extended', // The task list ID. See the TaskList section below for more information.
  new MyTask(
    $task_lists::get_list( 'extended' ), // The task list object.
  )
);
```

### Step 2 – Register the task in JavaScript.

Next, you have to add your task to the tasks list in JavaScript.

```jsx
import { __ } from '@wordpress/i18n';
import {
  WooOnboardingTask,
  WooOnboardingTaskListItem,
} from '@woocommerce/onboarding';

const Task = ( { onComplete, task, query } ) => {
  // Implement your task UI/feature here.
  return (
    <div>
    </div>
  );
};

registerPlugin( 'add-task-content', {
  render: () => (
    <WooOnboardingTask id="my-task">
      { ( {
        onComplete,
        query,
        task,
      } ) => <Task onComplete={ onComplete } task={ task } query={ query } /> }
    </WooOnboardingTask>
  )
```

### Example

> You can find an full example of how to add a custom task in the [examples directory](../examples/extensions/add-task/) as a WP plugin.

Reviewer: ```suggestion
You can find a complete example of how to add a custom task as a WordPress plugin in the [examples directory](../examples/extensions/add-task/).
```

## Models and classes

### TaskLists

> The `TaskLists` class acts as a data store for tasks. It allows you to create, initialize, add tasks, retrieve task lists, and perform other operations related to task management.

Reviewer: ```suggestion
The `TaskLists` class serves as a data store for tasks, providing functionality to create, initialize, add tasks, retrieve task lists, and perform other task management operations.
```

#### Methods

- `TaskLists::instance()`: Returns the class instance of the `TaskLists` interface.
- `TaskLists::init()`: Initializes the task lists. This method should be called to set up the necessary configurations and hooks for task management.

> - `TaskLists::is_experiment_treatment($name)`: Checks if an experiment is the treatment or control.

Reviewer: I'm a bit hesitant to add this since it's only useful internally, perhaps we could add more info
```suggestion
- `TaskLists::is_experiment_treatment($name)`: Checks if an experiment is the treatment or control. This is internally used by Woo.
```

    }

    public function get_title() {
        // Return the title of your task
    }

    public function get_content() {
        // Return the content/explanation of your task
    }

    public function get_time() {
        // Return the estimated time to complete the task
    }
    // Implement other abstract methods as needed
}

// Add MyTask to "test-list" task list
TaskLists::add_task(
  'test-list',
  new MyTask(
    TaskLists::get_list( 'test-list' )
  )
);
```
### Methods

- `$task->get_id(): string`: Returns the ID of the task.
- `$task->get_title(): string`: Returns the title of the task.
- `$task->get_content(): string`: Returns the content of the task.
- `$task->get_time(): string`: Returns the estimated time to complete the task.
- `$task->get_parent_id(): string`: Returns the ID of the parent task list.
- `$task->get_parent_options(): array`: Returns the options of the parent task list.
- `$task->get_parent_option($option_name): mixed|null`: Returns the value of a specific option from the parent task list.
- `$task->prefix_event($event_name): string`: Returns the event name prefixed with the task list's event prefix.

> - `$task->get_additional_info(): string`: Returns additional information about the task.
> - `$task->get_additional_data(): mixed|null`: Returns additional data associated with the task.

Reviewer: I think it'll be good for us to clarify the difference between these two.

```suggestion
- `$task->get_additional_info(): string`: Returns additional information about the task. Typically includes details, notes, or instructions related to the task itself.
- `$task->get_additional_data(): mixed|null`: Returns additional data associated with the task. It can be any type of data, such as arrays, objects, or simple values.
```
